src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -135,23 +135,41 @@ private async Task<TagHelperBinding> GetOriginTagHelperBindingAsync(DocumentSnap
                 return null;
             }
 
-            var node = owner.Ancestors().FirstOrDefault(n => n.Kind == SyntaxKind.MarkupTagHelperStartTag);
-            if (node == null || !(node is MarkupTagHelperStartTagSyntax tagHelperStartTag))
+            var node = owner.Ancestors().FirstOrDefault(n =>
+                n.Kind == SyntaxKind.MarkupTagHelperStartTag ||
+                n.Kind == SyntaxKind.MarkupTagHelperEndTag);
+            if (node is null)
             {
                 return null;
             }
 
-            if (!tagHelperStartTag.Name.Span.Contains(location.AbsoluteIndex))
+            var name = GetStartOrEndTagName(node);
+            if (name is null)
             {
                 return null;
             }
 
-            if (!(tagHelperStartTag?.Parent is MarkupTagHelperElementSyntax tagHelperElement))
+            if (!name.Span.Contains(location.AbsoluteIndex))
+            {
+                return null;
+            }
+
+            if (!(node.Parent is MarkupTagHelperElementSyntax tagHelperElement))
             {
                 return null;
             }
 
             return tagHelperElement.TagHelperInfo.BindingResult;
         }
+
+        private static SyntaxNode GetStartOrEndTagName(SyntaxNode node)
+        {
+            return node switch
+            {
+                MarkupTagHelperStartTagSyntax tagHelperStartTag => tagHelperStartTag.Name,
+                MarkupTagHelperEndTagSyntax tagHelperEndTag => tagHelperEndTag.Name,
```

Comment:
Looks like you expanded the logic to handle a few more cases. 👏 Love that you took the initiative to improve this even more!

Would you mind updating the tests to account for the new scenario?

Reply:
> Would you mind updating the tests to account for the new scenario?

So we don't actually have *any* testing for RazorDefinitionEndpoint. I can add it in, but wanted to confirm whether we're still going the moq'ing route, or not. This endpoint uses three new services whose integration with testing directly with the language server may not be a small task.

Reply:
I'd make the method internal for testing and given it's static you shouldn't need to use those services. That way you could unit test

Reply:
Thanks, updated!